annotations/relations.py

```diff
@@ -521,3 +522,86 @@ def update_relationset(template, raw_data, creator, text, project_id=None, relat
         relationset.save()
     return relationset
 
+def _create_relation(template_part, provided, relationset, cache={}, appellation_cache={}, project_id=None, creator=None, text=None, required=None, _as_key=None):
+    relation_cache = _get_cache(template_part, cache)
+    if relation_cache != None:
+        return relation_cache
+
+    field_handlers = _get_field_handlers(appellation_cache, project_id, creator, text, required, _as_key)
+    relation_data = _get_relation_data(relationset, creator, text)
+    _set_relation_data(template_part, provided, relationset, relation_data, field_handlers, inspect.currentframe().f_code.co_name, cache, appellation_cache, project_id, creator, text)
+    relation = Relation.objects.create(**relation_data)
+
+    if cache != None:
+        cache[template_part.id] = relation
+    return relation
+
+def _update_relation(template_part, provided, relationset, cache={}, appellation_cache={}, project_id=None, creator=None, text=None, required=None, _as_key=None):
+    relation_cache = _get_cache(template_part, cache)
+    if relation_cache != None:
+        return relation_cache
+
+    field_handlers = _get_field_handlers(appellation_cache, project_id, creator, text, required, _as_key)
+    relation_data = _get_relation_data(relationset, creator, text)
+    _set_relation_data(template_part, provided, relationset, relation_data, field_handlers, inspect.currentframe().f_code.co_name, cache, appellation_cache, project_id, creator, text)
+
+    relation_id_q = relationset.constituents.all().values('id')
```

Comment:
It looks like all you need is the first relation, is that right? If so, is there a need to get all ids? Can't you just do `relationset.constituents.all().first()`?